tests/vm.rs

```diff
@@ -557,10 +559,14 @@ fn run_expr_int() {
 
     let text = r#"io.run_expr "123" "#;
     let mut vm = make_vm();
-    let (result, _) = Compiler::new().run_io_expr::<String>(&mut vm, "<top>", text).unwrap();
-    let expected = "123 : Int";
-
-    assert_eq!(result, expected);
+    let (result, _) = Compiler::new().run_io_expr::<IO<String>>(&mut vm, "<top>", text).unwrap();
```

Comment:
Not really, its just a legacy quirk. Never had a reason not to run `IO` before and the current way of inspecting and running it is more re less a hack I guess. I think I will try and just have `run_expr` and then a secondary function which can run the `IO` action that is returned.